imjoy/imjoyPluginEngine.py

```diff
@@ -1014,26 +1014,33 @@ def scandir(path, type=None, recursive=False):
     if sid not in registered_sessions:
         logger.debug("client %s is not registered.", sid)
         return {"success": False, "error": "client has not been registered."}
-    workspace_dir = os.path.join(WORKSPACE_DIR, registered_sessions[sid]["workspace"])
 
-    path = kwargs.get("path", workspace_dir)
+    try:
+        workspace_dir = os.path.join(
+            WORKSPACE_DIR, registered_sessions[sid]["workspace"]
+        )
 
-    if not os.path.isabs(path):
-        path = os.path.join(workspace_dir, path)
-    path = os.path.normpath(os.path.expanduser(path))
+        path = kwargs.get("path", workspace_dir)
 
-    type = kwargs.get("type", None)
-    recursive = kwargs.get("recursive", False)
-    files_list = {"success": True}
-    files_list["path"] = path
-    files_list["name"] = os.path.basename(os.path.abspath(path))
-    files_list["type"] = "dir"
-    files_list["children"] = scandir(files_list["path"], type, recursive)
+        if not os.path.isabs(path):
+            path = os.path.join(workspace_dir, path)
+        path = os.path.normpath(os.path.expanduser(path))
+
+        type = kwargs.get("type", None)
+        recursive = kwargs.get("recursive", False)
+        files_list = {"success": True}
+        files_list["path"] = path
+        files_list["name"] = os.path.basename(os.path.abspath(path))
+        files_list["type"] = "dir"
+        files_list["children"] = scandir(files_list["path"], type, recursive)
 
-    if sys.platform == "win32" and os.path.abspath(path) == os.path.abspath("/"):
-        files_list["drives"] = get_drives()
+        if sys.platform == "win32" and os.path.abspath(path) == os.path.abspath("/"):
+            files_list["drives"] = get_drives()
 
-    return files_list
+        return files_list
+    except Exception as e:
```

Comment:
Do we need to put the except block down here? What calls can blow up? It's good practice to wrap as little as possible in the try... except. We can eg use try... except... else if needed.

Reply:
Calling scandir can fail due to non-existing path for example.

Reply:
Calling scandir can fail due to non-existing path for example.

Reply:
I would only wrap the line I know can error, and if we know the error we should catch that explicitly.